### PR TITLE
PyTorch MultiheadAttention API

### PR DESCRIPTION
**Pull request by ksivaman**

### CODE REVIEW

**Comment by ksivaman** [comment]
Pipeline 9426641

**Comment by ksivaman** [comment]
PTAL @timmoon10 @ptrendx. The key changes include:

1. Generating the documentation for the existing parameters and exposing the MHA API in `te.pytorch`.
2. Renaming `MultiHeadAttention` &rarr; `MulitheadAttention`, similar to PyTorch.
3. Adding the `return_bias` argument to the API since by default `TransformerLayer` always uses True in order to fuse with dropout and residual add, but as a standalone API we want to actually include the bias addition in the forward itself, not just parameter.
4. Setting default values for all arguments except `hidden_size` and `num_attention_heads`, similar to `torch.nn.MultiheadAttention`.
5. Adding numerics tests to compare with framework API.

**Review by timmoon10 — APPROVED** [review]
LGTM
